runtime/services/catalog/artifacts/yaml/yaml.go

```diff
@@ -8,7 +8,7 @@ import (
 
 	"github.com/rilldata/rill/runtime/drivers"
 	"github.com/rilldata/rill/runtime/services/catalog/artifacts"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
```

Comment:
yaml.v2 deserializes nested maps as `map[interface{}]interface{}` instead of `map[string]interface{}`
https://github.com/go-yaml/yaml/issues/139